checkov/bitbucket_pipelines/runner.py

```diff
@@ -0,0 +1,135 @@
+        self.add_root_image(file_path, images, root_image, workflow_line_numbers)
+
+        pipelines = workflow.get("pipelines", {})
+        default_pipeline = pipelines.pop("default", {})
+        self.add_default_pipeline_images(file_path, images, default_pipeline)
+
+        self.add_pipeline_images(file_path, images, pipelines)
+
+        return images
+
+    def add_root_image(self, file_path: str, images: [], root_image: str, workflow_line_numbers: dict) -> None:
+        if root_image:
+            for line_number, line_txt in workflow_line_numbers:
+                if "image" in line_txt and not line_txt.startswith(' '):
+                    start_line = line_number
+                    end_line = line_number
+                    image_id = self.pull_image(root_image)
+                    image_obj = Image(file_path=file_path, name=root_image, image_id=image_id,
+                                      start_line=start_line,
+                                      end_line=end_line)
+                    images.add(image_obj)
+
+    def add_pipeline_images(self, file_path: str, images: [], pipelines: dict):
+        for pipeline_name, pipeline_obj in pipelines.items():
+            if isinstance(pipeline_obj, dict):
+                for step_name, step_obj in pipeline_obj.items():
+                    if isinstance(step_obj, list):
+                        for step in step_obj:
+                            step_obj_def = step.get("step", {})
+                            image = step_obj_def.get("image", '')
+                            if image:
+                                image_id = self.pull_image(image)
+                                if image_id:
+                                    start_line = step_obj_def['__startline__']
+                                    end_line = step_obj_def['__endline__']
+
+                                    image_obj = Image(file_path=file_path, name=image, image_id=image_id,
+                                                      start_line=start_line,
+                                                      end_line=end_line)
+                                    images.add(image_obj)
+
+    def add_default_pipeline_images(self, file_path: str, images: [], default_pipeline: dict):
+        for step in default_pipeline:
+            if isinstance(step, dict):
+                step_obj = step.get('step', {})
+                image = step_obj.get("image", '')
+                if image:
+                    image_id = self.pull_image(image)
+                    if image_id:
+                        start_line = step_obj['__startline__']
+                        end_line = step_obj['__endline__']
+
+                        image_obj = Image(file_path=file_path, name=image, image_id=image_id,
+                                          start_line=start_line,
+                                          end_line=end_line)
+                        images.add(image_obj)
```

Comment:
this looks very similar to the previous method, probably you can abstract some parts